index.js

```diff
@@ -9,6 +10,40 @@ const nextButton = document.querySelector('[data-js="button-next"]');
 const pagination = document.querySelector('[data-js="pagination"]');
 
 // States
-const maxPage = 1;
-const page = 1;
-const searchQuery = "";
+let maxPage = 1;
+let page = 1;
+let searchQuery = "";
+
+async function fetchCharacters() {
+  try {
+    const response = await fetch(
+      `https://rickandmortyapi.com/api/character?page=${page}&name=${searchQuery}`
+    );
+    const { info, results } = await response.json();
+    maxPage = info.pages;
+    pagination.textContent = `${page} / ${info.pages}`;
+
+    results.forEach(createCharacterCard);
+  } catch (error) {
+    console.error("An error occurred while fetching characters:", error);
+  }
+}
+
+prevButton.addEventListener("click", () => {
+  page > 1 && page--;
+  fetchCharacters();
+});
+nextButton.addEventListener("click", () => {
+  page < maxPage && page++;
+  fetchCharacters();
+});
+
+searchBar.addEventListener("submit", (event) => {
+  event.preventDefault();
+  console.log("Search form submitted");
+  searchQuery = event.target.elements.query.value;
+  console.log("Search query:", searchQuery);
+  fetchCharacters();
+});
```

Comment:
Please remove all `console.log` before creating a PR. They are only used for development.
Additionally, you should set your `page=1` here. Otherwise your pagination won't show the page numbers correctly, when you use the searchbar on another page than page 1.